src/screens/TaskListScreen.tsx

```diff
@@ -0,0 +1,725 @@
+import React, { useState } from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  ScrollView,
+  TextInput,
+  Alert,
+  Modal,
+  Switch,
+} from 'react-native';
+import { useTaskContext } from '../context/TaskContext';
+import { Task } from '../types';
+
+const TaskListScreen: React.FC = () => {
+  const { tasks, addTask, updateTask, deleteTask, currentScores, getTopTasks, addProperty, answeredQuestions, questions } = useTaskContext();
+  const [showAddModal, setShowAddModal] = useState(false);
+  const [showAddPropertyModal, setShowAddPropertyModal] = useState(false);
+  const [showEditModal, setShowEditModal] = useState(false);
+  const [showFullResults, setShowFullResults] = useState(false);
+  const [newTaskName, setNewTaskName] = useState('');
+  const [newPropertyName, setNewPropertyName] = useState('');
+  const [newPropertyQuestion, setNewPropertyQuestion] = useState('');
+  const [editingTask, setEditingTask] = useState<Task | null>(null);
+  const [editTaskName, setEditTaskName] = useState('');
+  const [editTaskProperties, setEditTaskProperties] = useState<Record<string, boolean>>({});
+  const [newTaskProperties, setNewTaskProperties] = useState<Record<string, boolean>>({
+    indoor: false,
+    physical: false,
+    quick: false,
+    creative: false,
+    social: false,
+    learning: false,
+  });
+
+  const handleAddTask = () => {
+    if (!newTaskName.trim()) {
+      Alert.alert('Error', 'Please enter an activity name');
+      return;
+    }
+
+    const newTask = {
+      id: Date.now().toString(),
+      name: newTaskName.trim(),
+      properties: newTaskProperties,
+      score: 0,
+    };
+
+    addTask(newTask);
+    setNewTaskName('');
+    setNewTaskProperties({
+      indoor: false,
+      physical: false,
+      quick: false,
+      creative: false,
+      social: false,
+      learning: false,
+    });
+    setShowAddModal(false);
+  };
+
+  const handleAddProperty = () => {
```

Comment:
note: this has side effect of adding default false property to all tasks. is that clear enough?